### ISSUE TITLE
eth_getBlockByNumber panic with RlpExpectedToBeData

### ISSUE THREAD

**Issue by nanopool** [paritytech/parity#8881]
Parity 1.10.6, 1.12
Linux
built from source and pre-compiled version tested,  synchronized,  eth and etc chains tested

Sending a valid 'eth_getBlockByNumber' rpc request kills parity with panic.
Example request:
`curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1b4", true],"id":1}' -H "Content-Type: application/json" http://localhost:[rpc_port]`

Parity console output:

```
2018-06-12 20:16:02  Starting Parity/v1.10.6-stable-bc0d134-20180605/x86_64-linux-gnu/rustc1.26.1
2018-06-12 20:16:02  Keys path /root/.local/share/io.parity.ethereum/keys/Ethereum Classic
2018-06-12 20:16:02  DB path /root/.local/share/io.parity.ethereum/chains/classic/db/906a34e69aec8c0d
2018-06-12 20:16:02  Path to dapps /root/.local/share/io.parity.ethereum/dapps
2018-06-12 20:16:02  State DB configuration: fast
2018-06-12 20:16:02  Operating mode: active
2018-06-12 20:16:04  Configured for Ethereum Classic using Ethash engine
2018-06-12 20:16:11  Public node URL: enode://899780de0eb4a85cc6f90ae112c840469cf4ec4fab852ffeecfa6931b07ac50a530ef0352ffa0d155e137a140e5caad367615c89e5beaf5fd5718ce3f739f334@95.85.29.252:30303
2018-06-12 20:16:15  Imported #5986580 9652…2caf (2 txs, 0.05 Mgas, 13.40 ms, 0.75 KiB)

====================

stack backtrace:
   0:     0x560fdcd1481c - <no info>

Thread '<unnamed>' panicked at 'Unexpected rlp error: RlpExpectedToBeData', libcore/result.rs:945

This is a bug. Please report it at:

    https://github.com/paritytech/parity/issues/new
```

**Comment by nanopool** [comment]
It's the node with 1.9.7 version running without the issue. 
```
2018-06-13 14:42:47  Starting Parity/v1.9.7-unstable-7919629-20180423/x86_64-linux-gnu/rustc1.26.0
2018-06-13 14:42:47  Keys path /root/.local/share/io.parity.ethereum/keys/Ethereum Classic
2018-06-13 14:42:47  DB path /root/.local/share/io.parity.ethereum/chains/classic/db/906a34e69aec8c0d
2018-06-13 14:42:47  Path to dapps /root/.local/share/io.parity.ethereum/dapps
2018-06-13 14:42:47  State DB configuration: fast
2018-06-13 14:42:47  Operating mode: active
2018-06-13 14:42:48  Configured for Ethereum Classic using Ethash engine
```
The problem appears when I replace parity binary with a newer version. The old one is still working without an issue. Could it be I need to update anything else but 'parity' binary file?

**Comment by dvdplm** [comment]
@nanopool can you share what your configuration looks like please (command line options or config file)?

**Comment by nanopool** [comment]
It's quite simple:
`/root/bin/parity --chain=classic --rpcaddr 0.0.0.0 --jsonrpc-hosts="all" 2>&1 | tee -a /root/log/parity-$now.log`

The strange thing is I've found a random block number that works: "0x345566". None of the nearest blocks works though. Neigher "0x345565" nor "0x345567".

I've tested on my home linux machine and it works with the new binary version. Will upload the blockchain to the node from home pc and test it again.
